safe_control_gym/controllers/base_controller.py

```diff
@@ -1,112 +1,106 @@
-                 checkpoint_path="temp/model_latest.pt",
-                 output_dir="temp",
+                 checkpoint_path='temp/model_latest.pt',
+                 output_dir='temp',
                  use_gpu=False,
                  seed=0,
                  **kwargs
                  ):
-        """Initializes controller agent.
+        '''Initializes controller agent.
 
         Args:
-            env_func (callable): function to instantiate task/env.
-            training (bool): training flag.
-            checkpoint_path (str): file to save trained model & experiment state.
-            output_dir (str): folder to write outputs.
+            env_func (callable): Function to instantiate task/env.
+            training (bool): Training flag.
+            checkpoint_path (str): File to save trained model & experiment state.
+            output_dir (str): Folder to write outputs.
             use_gpu (bool): False (use cpu) True (use cuda).
-            seed (int): random seed.
+            seed (int): Random seed.
+        '''
 
-        """
         # Base args.
         self.env_func = env_func
         self.training = training
         self.checkpoint_path = checkpoint_path
         self.output_dir = output_dir
         self.use_gpu = use_gpu and torch.cuda.is_available()
-        self.device = 'cpu' if self.use_gpu == False else 'cuda'
+        self.device = 'cpu' if self.use_gpu is False else 'cuda'
         self.seed = seed
         # Algorithm specific args.
-        for k, v in kwargs.items():
-            self.__dict__[k] = v
-        
+        for key, value in kwargs.items():
+            self.__dict__[key] = value
+
         self.setup_results_dict()
 
     def select_action(self, obs, info=None):
-        """Determine the action to take at the current timestep.
+        '''Determine the action to take at the current timestep.
 
         Args:
-            obs (np.array): the observation at this timestep
-            info (list): the info at this timestep
-        
+            obs (np.array): The observation at this timestep.
+            info (dict): The info at this timestep.
+
         Returns:
-            action (np.array): the action chosen by the controller
-        """
-        return 
+            action (np.array): The action chosen by the controller.
+        '''
+        return
 
     def learn(self,
               env=None,
               **kwargs
               ):
-        """Performs learning (pre-training, training, fine-tuning, etc).
+        '''Performs learning (pre-training, training, fine-tuning, etc).
 
         Args:
-            env (gym.Env): the environment to be used for training
-        """
-        pass
+            env (gym.Env): The environment to be used for training.
+        '''
+        return
 
     def reset(self):
-        """Do initializations for training or evaluation.
-
-        """
-        pass
+        '''Do initializations for training or evaluation. '''
+        return
```

Comment:
shouldn't this be something like 
```
raise NotImplementedError
```
if we want to make re-implementation mandatory?

Reply:
You're right, fixed it. Also made them abstract methods in the other PR.